ui/src/components/input/__tests__/QInput.cy.js

```diff
@@ -1,92 +1,773 @@
+              rules: [ (value) => !!value || '' ]
+            }
+          })
+          getHostElement().get('.q-field__messages').should('exist')
+
+          mountQInput({
+            props: {
+              rules: [ (value) => !!value || '' ],
+              hideBottomSpace: true
+            }
+          })
+
+          getHostElement().get('.q-field__messages').should('not.exist')
+        })
+      })
+
+      describe('(prop): rounded', () => {
+        it('should use a rounded design', () => {
+          mountQInput({
+            props: {
+              rounded: true
+            }
+          })
+
+          getHostElement().get('.q-field--rounded').should('exist')
+        })
+      })
+
+      describe('(prop): square', () => {
+        it('should use a square design', () => {
+          mountQInput({
+            props: {
+              square: true
+            }
+          })
+
+          getHostElement().get('.q-field--square').should('exist')
+        })
+      })
+
+      describe('(prop): dense', () => {
+        it('should use a dense design', () => {
+          mountQInput({
+            props: {
+              dense: true
+            }
+          })
+
+          getHostElement().get('.q-field--dense').should('exist')
         })
       })
     })
   })
 
   describe('Events', () => {
     describe('(event): update:model-value', () => {
-      it.skip(' ', () => {
-        //
+      it('should emit onUpdate:modelValue event', () => {
+        const fn = cy.stub()
+        const text = 'Hello there'
+        mountQInput({
+          props: {
+            'onUpdate:modelValue': fn
+          }
+        })
+        getHostElement()
+          .get('input')
+          .type(text)
+          .then(() => {
+            expect(fn).to.be.calledWith(text)
+          })
       })
     })
 
     describe('(event): focus', () => {
-      it.skip(' ', () => {
-        //
+      it('should emit focus event', () => {
+        const fn = cy.stub()
+        mountQInput({
+          props: {
+            onfocus: fn,
+            autoFocus: true
+          }
+        })
+
+        getHostElement()
+          .get('input')
+          .as('input')
+          .focus()
+
+        cy.get('@input').then(() => {
+          expect(fn).to.be.calledWith()
+        })
       })
     })
 
     describe('(event): blur', () => {
-      it.skip(' ', () => {
-        //
+      it('should emit blur event', () => {
+        const fn = cy.stub()
+        mountQInput({
+          props: {
+            onblur: fn
+          }
+        })
+
+        getHostElement()
+          .get('input')
+          .focus()
+          .blur()
+          .then(() => {
+            expect(fn).to.be.calledWith()
+          })
       })
     })
   })
 
   describe('Methods', () => {
     describe('(method): focus', () => {
-      it.skip(' ', () => {
-        //
+      it('should focus the component', () => {
+        mountQInput()
+
+        getHostElement()
```

Comment:
I'd separate those kind of assertions in multiple lines, even calling again `getHostElement()`
It's less performant, but more maintenable and readable, and we saw many problems between Cypress versions while using chaining too much

It's also confusing as a reader as you'd expect the input to be searched inside `.q-field--focused`, while you're searching it into the host element